cognito-post-authentication/src/main/java/no/unit/nva/cognito/service/UserService.java

```diff
@@ -90,18 +95,36 @@ private String toRoleGroupName(Role role) {
         return String.format(ROLE_GROUP_TEMPLATE, role.getRolename());
     }
 
-    private User createUser(String feideId, String customerId, String affiliation) {
-        List<Role> roles = createRoles(affiliation);
-        User user = new User(feideId, customerId, roles);
+    private User createUser(String username, Optional<String> customerId, String affiliation) {
+        User user;
+        if (customerId.isPresent()) {
```

Comment:
Your code is maybe more readable so the following suggestion is optional

Suggestion:

```
User user = customerId.map(s -> createUserForInstitution(username, s, affiliation))
            .orElseGet(() -> createUserWithoutInstitution(username));
```